release-guide.md

```diff
@@ -0,0 +1,49 @@
+# Release Guide
```

Comment:
I thought about it. And, figured it's better to keep everything in one-place. 

I don't mind moving this to our wiki.